packages/flutter/lib/src/material/input.dart

```diff
@@ -84,9 +84,17 @@ class _InputState extends State<Input> {
     bool focused = focusContext != null && Focus.at(focusContext, autofocus: config.autofocus);
 
     TextStyle textStyle = config.style ?? themeData.text.subhead;
-    Color focusHighlightColor = themeData.accentColor;
-    if (themeData.primarySwatch != null)
-      focusHighlightColor = focused ? themeData.primarySwatch[400] : themeData.hintColor;
+    Color activeColor = themeData.hintColor;
+    if (focused) {
+      switch (themeData.brightness) {
```

Comment:
we have this switch all over. maybe we should consider brightness to be an input too. that would solve several problems e.g. in lerping.